atlassian_jwt_auth/key.py

```diff
@@ -73,18 +73,21 @@ class HTTPSPublicKeyRetriever(BasePublicKeyRetriever):
          given key id.
     """
 
-    def __init__(self, base_url):
+    def __init__(self, base_url, retry=3):
```

Comment:
This should cover all of the feedback:

> I'd like to suggest a retry count of ~ 3 with a lower back-off interval time [...]

and

> I think that retry count should be parameterized as well so that consumers can opt-out if they wish.

The commit message spells out how to customize retries further.